### PR TITLE
tools,meta: make AUTHORS reflect all the contributors

### PR DESCRIPTION
**Pull request by aqrln**
Right now we only have those who have made valuable contributions noted in the AUTHORS file. But there are people who have made minor contributions, like fixing typos, updating the links and such, to examples and README.md (which is effectively a part of documentation, and documentation is a part of the software, according to the MIT license). And however small their contributions may be, these are still contributions and it is fair to add those people to the AUTHORS file.

Additionally, the process of updating this file is now automated.

#### Changes:

* Add new `update-authors.js` tool that traverses commit history using `git log` and creates authors list.
* Decouple common functions into a module shared between tools.
* Add `.mailmap` file with proper names and emails of active collaborators.  This file is automatically parsed by Git so that `git log --format='%aN <%aE>'` won't show duplicate entries in case something was committed with misconfigured Git.  E.g., without this file @tshemsedinov appears twice in the list, as Timur Shemsedinov and as tshemsedinov, but with mailmap applied Git automatically coalesces these entries to use the full name only.
* Update the list using `tools/update-authors.js`.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -1,3 +1,6 @@
 Timur Shemsedinov <timur.shemsedinov@gmail.com>
+Dimon Durak <Dimon Durak>
```
Comment (aqrln) [inline AUTHORS]: @dimon-durak can you please provide your real name and email if you want to be included to the list?

File: AUTHORS [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+Artem Chernenkiy <notthewhite@gmail.com>
 Alexey Orlenko <eaglexrlnk@gmail.com>
 Mykola Bilochub <nbelochub@gmail.com>
+Vlad <Dzyubavlad@gmail.com>
```
Comment (aqrln) [inline AUTHORS]: @DzyubSpirit please check if you email is correct here. And how would you prefer your full name to be spelled?

Comment (DzyubSpirit) [inline AUTHORS]: Email is correct and I prefer to be Dzyuba WhatIsLove.

Thanks.

Comment (aqrln) [inline AUTHORS]: @DzyubSpirit I think we prefer real names unless you have strong reasons to not be called Vlad Dzyuba. I was asking about the correct spelling of your last name, actually :)
What about email, maybe I should have been more explicit too: the question is whether it should be written with a capital letter or I can lowercase it.

Comment (DzyubSpirit) [inline AUTHORS]: @aqrln, okey. Right spelling of my last name is Dziuba. Let it be this way. My email case does not matter, so you can do what it is needed with it.

Comment (aqrln) [inline AUTHORS]: @DzyubSpirit done.

File: AUTHORS (hunk 1 above)
Comment (aqrln) [inline AUTHORS]: @dimon-durak ping

Comment (dimon-durak) [inline AUTHORS]: Dmitry Borisov
dimon.durak@gmail.com

Comment (aqrln) [inline AUTHORS]: Thanks, I've added an entry for you in `.mailmap` and regenerated the list of contributors.

**Comment by aqrln** [comment]
ping @belochub @tshemsedinov

**Comment by aqrln** [comment]
@tshemsedinov ping

**Comment by aqrln** [comment]
It's been a while and I suppose there are no objections. Landed in ebf9e2d, bf40c83 and 73d2533.

**Pull request closed without merging**